File: programming/typescript/1-basics/5 - loops.ts
/* Loops */

// for
for (let i = 0; i < 5; i++) {
  console.log(i) // 0, 1, 2, 3, 4
}

// for of
let fruits: string[] = ['apple', 'banana', 'orange']
for (let fruit of fruits) {
  console.log(fruit) // 'apple', 'banana', 'orange'
}

// for in
let person = { name: 'John', age: 30 }
for (let key in person) {
  console.log(key, person[key]) // 'name John', 'age 30'
}

// while
let count = 0
while (count < 3) {
  console.log(count) // 0, 1, 2
  count++
}

// do...while...
let x = 0
do {
  console.log(x) // 0
  x++
} while (x < 0)

File: programming/python/1-basics/5.loops.py
# Loops

# while loop
i = 0
while i < 10:
    if i == 5:
        break # stop the loop
    if i == 2:
        i += 1
        continue # stop the current iteration
    print(i)
    i += 1

# for loop (in)
fruits = ['apple', 'banana', 'kwi', 'coconut']

for fruit in fruits:
    if fruit == "banana":
        break # break the entry loop
    if fruit == "coconut":
        continue # skip the current interation
    print(fruit)

# for loop (in range)
for x in range(10):
    print(x)

# for loop (in range with start and end)
for x in range(2, 10):
    print(x)

# for loop (in range with start, end and step)
for x in range(2, 10, 2):
    print(x)
